src/linpred.jl

```diff
@@ -234,13 +234,24 @@ function show(io::IO, obj::LinPredModel)
     println(io, "$(typeof(obj)):\n\nCoefficients:\n", coeftable(obj))
 end
 
-modelframe(obj::LinPredModel) = obj.fr
+function modelframe(f::FormulaTerm, data, contrasts::AbstractDict)
+    Tables.istable(data) ||
+        throw(ArgumentError("expected data in a Table, got $(typeof(data))"))
+    t = Tables.columntable(data)
+    msg = StatsModels.checknamesexist(f, t)
+    msg != "" && throw(ArgumentError(msg))
+    data, _ = StatsModels.missing_omit(t, f)
+    sch = schema(f, data, contrasts)
+    f = apply_schema(f, sch, LinPredModel)
+    f, modelcols(f, data)
+end
+
 modelmatrix(obj::LinPredModel) = obj.pp.X
 response(obj::LinPredModel) = obj.rr.y
 
 fitted(m::LinPredModel) = m.rr.mu
 predict(mm::LinPredModel) = fitted(mm)
-StatsModels.formula(obj::LinPredModel) = modelframe(obj).formula
+formula(obj::LinPredModel) = obj.f
```

Comment:
Should we error on `isnothing(obj.f)` ? I suspect downstream consumers of `formula()` expect for it to return a `FormulaTerm`.